tools/wpt/tests/test_browser.py

```diff
@@ -196,37 +196,130 @@ def test_safari_version_errors(mocked_check_output):
     mocked_check_output.side_effect = subprocess.CalledProcessError(1, 'cmd')
     assert safari.version(webdriver_binary="safaridriver") is None
 
-@pytest.fixture
-def safari_downloads_page_stp_section():
-    file_path = os.path.join(
-        os.path.dirname(__file__),
-        'safari_downloads_page_stp_section.html')
-    with open(file_path) as fp:
-        return fp.read()
-
-@mock.patch('tools.wpt.browser.get')
-def test_safari_find_downloads_stp(mocked_get, safari_downloads_page_stp_section):
+
+@pytest.mark.parametrize(
+    "page_path",
+    sorted(
+        p.path
+        for p in os.scandir(os.path.join(os.path.dirname(__file__), "safari-downloads"))
+        if p.name.endswith(".html")
+    ),
+)
+@mock.patch("tools.wpt.browser.get")
+def test_safari_find_downloads_stp(mocked_get, page_path):
+    safari = browser.Safari(logger)
+
+    # Setup mock
```

Comment:
I think this `setup mock` code is now copied to enough places that it might be worth factoring out into a function, even in tests.

Reply:
Yeah, I debated this, but felt like it was just about short enough to not be worth it?